client/powerline.c

```diff
@@ -90,8 +91,18 @@ int main(int argc, char *argv[]) {
 	memset(&server, 0, sizeof(struct sockaddr_un));
 	server.sun_family = AF_UNIX;
 	strncpy(server.sun_path A, address, strlen(address));
+#ifdef __linux__
+	/* We use a socket in the abstract namespace on Linux, and because SUN_LEN uses strlen, it does not work for these. The struct size also fails, so compute manually. */
+	socklen = sizeof(server.sun_family) + strlen(address) A;
+#elif defined SUN_LEN
+	/* If the vendor provided SUN_LEN, we may as well use it. */
+	socklen = SUN_LEN(&server);
+#else
+	/* SUN_LEN is not POSIX, so if it was not provided, use the struct size as a fallback. */
+	socklen = sizeof(struct sockaddr_un);
+#endif
 
-	if (connect(sd, (struct sockaddr *) &server, (socklen_t) (sizeof(server.sun_family) + strlen(address) A)) < 0) {
+	if (connect(sd, (struct sockaddr *) &server, socklen) < 0) {
```

Comment:
Maybe we should just fix it once and for all [as suggested here](https://lists.debian.org/debian-glibc/2004/10/msg00135.html)? I'll give it a try.